Use integer comparison

UnwashedMeme opened the pull request:
As CoreOS version numbers are now greater than 1000 the lexical comparison fails; convert to integer and compare using that.

UnwashedMeme committed:
Use integer comparison

As CoreOS version numbers are now greater than 1000 the lexical comparison fails; convert to integer and compare using that.

mmoll commented:
:+1: 
Could you extract that to an `os_major` variable in the header, like it's some for some other OSes? That line is already quite long. :)

UnwashedMeme committed:
Split compound ExecStart into multiple lines

This was using bash `&&` to ensure consecutive success only commands. systemd 
has that built in so just use that directly.  This also tames the super long line to be much 
more readable.

UnwashedMeme commented:
I'll admit I might have been looking for a "Rolling-eyes" emoji at that request but this does read a hell of a lot easier. It is especially easier to try and edit inside foreman's in-browser editor so yeah that was a good call.

mmoll commented:
@UnwashedMeme personally I use the foreman_templates plugin to sync all the templates from this git repo here, so I'm rarely using that editor... ;)
@johscheuer please have a look

johscheuer commented on coreos/provision.erb:
```diff
@@ -14,7 +17,13 @@ coreos:
+        ExecStart=/usr/bin/coreos-install \
+          -C <%= @host.operatingsystem.release_name %> \
+          -d <%= @host.params['install-disk'] || '/dev/sda' %> \
+          -c /home/core/cloud-config.yml <% if os_major >= 557 -%>-b <%= @mediapath %><% end %>
```
@UnwashedMeme @mmoll is this really a Use Case to support such an old CoreOS version? [557.0](https://coreos.com/releases/#557.0.0) came out January 6, 2015 ...  with `Docker 1.3` + `etcd 0.4.6` personally I think we should just drop this part with `<% if os_major >= 557 -%>` if somebody really wants to use such an old version he just have to remove the part `-b <%= @mediapath %>`.

I mean you use CoreOS because you want a minimal (always) up to date Linux Distro (also after the reboot CoreOS would fetch the newest version of the channel and install this version (I fixed this one https://github.com/theforeman/community-templates/pull/294 )^^).

johscheuer commented:
👍 to the reformatting and update the systemd service :)

UnwashedMeme commented:
>  is this really a Use Case to support such an old CoreOS version?

It certainly isn't for me; I tend to agree that it's kind of against the point of CoreOS but, once fixed, the conditional wasn't bothering me and I figured leaving it in was less controversial.

mmoll commented:
It doesn't hurt to keep it, IMHO

mmoll commented:
merged, thanks @UnwashedMeme!
